h5netcdf/core.py

```diff
@@ -111,7 +111,19 @@ def __init__(self, parent, name, dimensions=None):
         self._parent_ref = weakref.ref(parent)
         self._root_ref = weakref.ref(parent._root)
         self._h5path = _join_h5paths(parent.name, name)
+
+        self._h5ds = self._root._h5file[self._h5path]
+        # fix name if _nc4_non_coord_
+        self._name = self._h5ds.name.replace("_nc4_non_coord_", "")
```

Comment:
Yes, but that was from the viewpoint of append/write. So that in any case the data/metadata is fetched from the underlying h5py/hdf5 resource. With respect to performance we surely can try to cache things, but need to think about how to do that correctly for append/write.

This `_nc4_non_coord_` is a somewhat special case, as in netcdf-c a variable and a coordinate with the same name can coexist. It is handled by prepending the underlying hdf5 name with `_nc4_non_coord_`.